### PR TITLE
Add dishwasher control cluster xml

### PR DESCRIPTION
**Pull request by rbultman**
This PR primarily adds the dishwasher control cluster XML prior to the start of the SDK work needed to implement the dishwasher device type for Matter 1.2.

### CODE REVIEW

**Comment by bzbarsky-apple** [comment]
@rbultman You need to run `zap_regen_all.py`, looks like?

**Comment by bzbarsky-apple** [comment]
> Should I be committing all of the C, java, and other files that get updated when zap-regen-all is run?

Yes.

> I assumed those were essentially "build" files and would not need to be committed.

There is some build-time codegen, but the files that are generated at build time are not committed at all, and are not present in the source tree at any point, generally.  The files you see getting modified when zap-regen-all is run need to be committed in the modified state; they are not generated at build-time.

File: src/app/zap-templates/zcl/data-model/chip/dishwasher-control-cluster.xml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+limitations under the License.
+-->
+<configurator>
+  <domain name="Appliances"/>
+
+  <bitmap name="DishwasherControlFeature" type="BITMAP32">
+    <cluster code="0x0058" />
+    <field name="STEAMWASH" mask="0x01" />
+    <field name="HEATEDDRY" mask="0x02" />
+    <field name="EXTDRY" mask="0x04" />
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/dishwasher-control-cluster.xml:24]: The field names should be the feature names, not the feature codes, to match other clusters.  So:
```suggestion
    <field name="SteamWash" mask="0x01" />
    <field name="HeatedDry" mask="0x02" />
    <field name="ExtendedDry" mask="0x04" />
```

File: src/app/zap-templates/zcl/data-model/chip/dishwasher-control-cluster.xml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    <name>Dishwasher Control</name>
+    <code>0x0058</code>
+    <define>DISHWASHER_CONTROL_CLUSTER</define>
+    <client init="false" tick="false">true</client>
+    <server init="false" tick="false">true</server>
+    <description>This cluster provides a way to access options associated with the operation of a Dishwasher device type.</description>
+
+    <globalAttribute side="server" code="0xFFFD" value="1" />
+
+    <attribute side="server" code="0x0000" define="AVAILABLE_OPTIONS_FOR_CURRENT_MODE" type="BITMAP32" entryType="INT32U" min="0x00" max="0x07"      writable="false" isNullable="false" optional="false">AvailableOptionsForCurrentMode</attribute>
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/dishwasher-control-cluster.xml:38]:
```suggestion
    <attribute side="server" code="0x0000" define="AVAILABLE_OPTIONS_FOR_CURRENT_MODE" type="BITMAP32" min="0x00" max="0x07"      writable="false" isNullable="false" optional="false">AvailableOptionsForCurrentMode</attribute>
```
Should there be an actual bitmap type defined for this in the spec (and here)?  But for now, matching the spec is the right thing.

**Comment by andy31415** [comment]
Android fails with 

```
INFO    2023-05-05 17:34:00 CRITICAL    'jni/DishwasherControlClient-ReadImpl.cpp' was generated but not expected
INFO    2023-05-05 17:34:00 CRITICAL    'jni/DishwasherControlClient-InvokeSubscribeImpl.cpp' was generated but not expected
```

you should add the new files into `src/controller/data_model/BUILD.gn`

**Comment by rbultman** [comment]
The dishwasher control cluster is being removed from the 1.2 ballot so this PR is being closed.

**Pull request closed without merging**